airflow/providers/google/cloud/hooks/compute_ssh.py

```diff
@@ -0,0 +1,250 @@
+            raise AirflowException(f'The Command {cmd} returned error status {return_code}')
+        if output:
+            self.log.info(output)
+
+    def create_ssh_key(self,
+                       account: str,
+                       expire_time: int = 300) -> str:
+        """
+        Generate a temporary SSH key and apply it to the specified account
+
+        :param account: The service account id
+        :type account: str
+        :param expire_time: The maximum amount of time before the private key is expired
+        :type expire_time: int
+        """
+        try:
+            self.log.info("Generating ssh keys...")
+            private_key_file = '/tmp/key-' + str(uuid.uuid4())
+            private_key_obj = paramiko.RSAKey.generate(2048)
+            private_key_obj.write_private_key_file(private_key_file)
+        except (IOError, SSHException) as err:
+            raise AirflowException(f"Error encountered creating ssh keys, {err}")
+
+        with open(private_key_file + '.pub', 'r') as original:
+            public_key = original.read().strip()
+
+        # Expiration time is in microseconds.
+        expiration = int((time.time() + expire_time) * 1000000)
+
+        body = {
+            'key': public_key,
+            'expirationTimeUsec': expiration,
+        }
+        self.get_conn(). \
+            users().importSshPublicKey(  # pylint: disable=no-member
+            parent=account, body=body).execute_in_compute()
+        return private_key_file
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def run(self,
+            cmd: str, instance: str, zone: str,
+            project_id: Optional[str] = None,
+            expire_time: int = 300,
+            hostname: Optional[str] = None,
+            strict_host_key_checking: bool = True
+            ) -> Any:
+        """
+         Execute a command in a remote instance
+
+        :param cmd: The command to execute on the instance
+        :type cmd: str
+        :param instance: The name of the instance the command will be executed on
+        :type instance: str
+        :param zone: The zone where the instance is located
+        :type zone: str
+        :param project_id: The project ID of the instance
+        :type project_id: Optional[str]
+        :param expire_time: The maximum amount of time before the private key is expired
+        :type expire_time: int
+        :param hostname: The hostname of the instance.
+        :type hostname: str
+        :param strict_host_key_checking: Whether to verify the host key
+        :type strict_host_key_checking: bool
+        """
+        hostname = self.get_hostname(project_id=project_id,
+                                     instance=instance,
+                                     zone=zone,
+                                     hostname=hostname)
+
+        account = self.get_service_account()
+
+        username = self.get_username()
+
+        private_key_file = self.create_ssh_key(account=account,
```

Comment:
It would be great if we also supported authorizations with instance metadata.

Reply:
Using project or instance metadata seem discouraged. See
https://cloud.google.com/compute/docs/instances/connecting-advanced#provide-key

What do you think?